src/System.IO.Compression.ZipFile/tests/ZipFileInvalidFileTests.cs

```diff
@@ -46,55 +46,85 @@ public void InvalidConstructors()
         [Fact]
         public void InvalidFiles()
         {
-            ConstructorThrows<InvalidDataException>(() => ZipFile.OpenRead(bad("EOCDmissing.zip")));
-            ConstructorThrows<InvalidDataException>(() => ZipFile.Open(bad("EOCDmissing.zip"), ZipArchiveMode.Update));
-            ConstructorThrows<InvalidDataException>(() => ZipFile.OpenRead(bad("CDoffsetOutOfBounds.zip")));
-            ConstructorThrows<InvalidDataException>(() => ZipFile.Open(bad("CDoffsetOutOfBounds.zip"), ZipArchiveMode.Update));
+            using (TempFile testArchive = CreateTempCopyFile(bad("EOCDmissing.zip"), GetTestFilePath()))
+            {
+                ConstructorThrows<InvalidDataException>(() => ZipFile.OpenRead(testArchive.Path));
```

Comment:
> Correct, which explains why the ConstructorThrows helper isn't necessary, 

Right

> We could argue that ZipArchive should clean up itself if it fails...

I've not looked at the code... if it's opening streams and then leaving them open in the case of failure, that's a bug.